pallets/funding/src/instantiator.rs

```diff
@@ -1339,6 +1338,209 @@ impl<
+			ProjectStatus::FundingSuccessful => Self::settle_successful_project(project_id),
+			ProjectStatus::FundingFailed | ProjectStatus::EvaluationFailed => Self::settle_failed_project(project_id),
+			_ => panic!("Project should be in FundingSuccessful, FundingFailed or EvaluationFailed status"),
+		})
+	}
+
+	fn settle_successful_project(project_id: ProjectId) -> Result<(), DispatchError> {
+		Evaluations::<T>::iter_prefix((project_id,))
+			.try_for_each(|(_, evaluation)| Pallet::<T>::do_settle_successful_evaluation(evaluation, project_id))?;
+
+		Bids::<T>::iter_prefix((project_id,))
+			.try_for_each(|(_, bid)| Pallet::<T>::do_settle_successful_bid(bid, project_id))?;
+
+		Contributions::<T>::iter_prefix((project_id,))
+			.try_for_each(|(_, contribution)| Pallet::<T>::do_settle_successful_contribution(contribution, project_id))
+	}
+
+	fn settle_failed_project(project_id: ProjectId) -> Result<(), DispatchError> {
+		Evaluations::<T>::iter_prefix((project_id,))
+			.try_for_each(|(_, evaluation)| Pallet::<T>::do_settle_failed_evaluation(evaluation, project_id))?;
+
+		Bids::<T>::iter_prefix((project_id,))
+			.try_for_each(|(_, bid)| Pallet::<T>::do_settle_failed_bid(bid, project_id))?;
+
+		Contributions::<T>::iter_prefix((project_id,))
+			.try_for_each(|(_, contribution)| Pallet::<T>::do_settle_failed_contribution(contribution, project_id))?;
+
+		Ok(())
+	}
+
+	pub fn get_evaluations(&mut self, project_id: ProjectId) -> Vec<EvaluationInfoOf<T>> {
+		self.execute(|| Evaluations::<T>::iter_prefix_values((project_id,)).collect())
+	}
+
+	pub fn get_bids(&mut self, project_id: ProjectId) -> Vec<BidInfoOf<T>> {
+		self.execute(|| Bids::<T>::iter_prefix_values((project_id,)).collect())
+	}
+
+	pub fn get_contributions(&mut self, project_id: ProjectId) -> Vec<ContributionInfoOf<T>> {
+		self.execute(|| Contributions::<T>::iter_prefix_values((project_id,)).collect())
+	}
+
+	// Used to check if all evaluations are settled correctly. We cannot check amount of
+	// contributions minted for the user, as they could have received more tokens from other participations.
+	pub fn assert_evaluations_settled(
+		&mut self,
+		project_id: ProjectId,
+		evaluations: Vec<EvaluationInfoOf<T>>,
+		percentage: u64,
+	) {
+		let details = self.get_project_details(project_id);
+		assert!(matches!(
+			details.status,
+			ProjectStatus::FundingSuccessful | ProjectStatus::FundingFailed | ProjectStatus::EvaluationFailed
+		));
+
+		self.execute(|| {
+			for evaluation in evaluations {
+				let reward_info =
+					ProjectsDetails::<T>::get(project_id).unwrap().evaluation_round_info.evaluators_outcome;
+				let account = evaluation.evaluator.clone();
+				assert_eq!(Evaluations::<T>::iter_prefix_values((&project_id, &account)).count(), 0);
+				let reserved = <T as Config>::NativeCurrency::balance_on_hold(
+					&(HoldReason::Evaluation(project_id).into()),
+					&account,
+				);
+				assert_eq!(reserved, 0u64.into());
+
+				match percentage {
```

Comment:
Not sure if that makes sense, as we already use this percentage variable for creating the project, so we know that the project has this percentage. I think it would over complicate this function.